app/assets/javascripts/discourse/app/components/relative-time.js

```diff
@@ -0,0 +1,82 @@
+import discourseComputed, { on } from "discourse-common/utils/decorators";
+
+import Component from "@ember/component";
+import I18n from "I18n";
+import { action } from "@ember/object";
+
+export default Component.extend({
+  tagName: "",
+  selectedInterval: "mins",
+  durationMinutes: null,
+
+  @on("init")
+  cloneDuration() {
+    let mins = this.durationMinutes;
+
+    if (mins >= 1440) {
+      this.setProperties({
+        _duration: Math.floor(mins / 60 / 24),
+        selectedInterval: "days",
+      });
+    } else if (mins >= 60) {
+      this.setProperties({
+        _duration: Math.floor(mins / 60),
+        selectedInterval: "hours",
+      });
+    } else {
+      this.setProperties({
+        _duration: mins,
+        selectedInterval: "mins",
+      });
+    }
+  },
+
+  @discourseComputed
+  intervals() {
+    return [
+      { id: "mins", name: I18n.t("relative_time.minutes") },
+      { id: "hours", name: I18n.t("relative_time.hours") },
+      { id: "days", name: I18n.t("relative_time.days") },
+      { id: "months", name: I18n.t("relative_time.months") },
+    ];
+  },
+
+  @discourseComputed("selectedInterval", "_duration")
```

Comment:
This is not well documented but we have a convention in Discourse that `_underscored` variable names are used for private members that are not computed or observed. Since `_duration` is used in a computed property I'd recommend changing it to `duration`.